app/services/action_handlers/pull_request.rb

```diff
@@ -26,16 +26,30 @@ def closed
     end
 
     def review_request_removed
-      reviewer = User.find_by!(github_id: @payload['pull_request']['requested_reviewers']
-                                            .first['id'])
-      @entity.review_requests.find_by!(reviewer: reviewer).removed!
+      pr_data = @payload['pull_request']
+      reviewers_github_id = pr_data['requested_reviewers'].map { |reviewer| reviewer['id'] }
+      owner_github_id = pr_data['user']['id']
+      owner = User.find_by!(github_id: owner_github_id).id
+      remove_review_requests(owner, reviewers_github_id)
     end
 
     def review_requested
       pr_data = @payload['pull_request']
       owner = find_or_create_user(pr_data['user'])
-      reviewer = find_or_create_user(pr_data['requested_reviewers'].first)
-      @entity.review_requests.create!(owner: owner, reviewer: reviewer)
+      pr_data['requested_reviewers'].each do |raw_reviewer|
+        reviewer = find_or_create_user(raw_reviewer)
+        @entity.review_requests.create!(owner: owner, reviewer: reviewer, state: 'active')
+      rescue ActiveRecord::RecordInvalid => e
+        raise unless e.message == 'Validation failed: Pull request has already been taken'
+      end
+    end
+
+    def remove_review_requests(owner, reviewers)
+      @entity.review_requests
+             .includes(:owner)
+             .where.not(reviewer: User.where(github_id: reviewers))
+             .where(owner: owner)
```

Comment:
Why do we have to check for the `owner` here? `@entity` is the PR, right? That means we already know that the review requests belong to the PR, and therefore the owner.